Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ViewBookmarkLoader.cpp

```diff
@@ -0,0 +1,552 @@
+
+                if (isMerged)
+                {
+                    using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+                    AZStd::string bookmarkKey = s_viewBookmarksRegistryPath + m_bookmarkfileName;
+                    auto viewBookmarkSettingsKey = FixedValueString::format(bookmarkKey.c_str());
+                    ViewBookmarkVisitor viewBookmarkVisitor;
+
+                    const bool visitedViewBookmarks = registry->Visit(viewBookmarkVisitor, viewBookmarkSettingsKey);
+
+                    if (visitedViewBookmarks)
+                    {
+                        m_localBookmarks = viewBookmarkVisitor.m_bookmarkMap.at(m_bookmarkfileName);
+                        m_localBookmarkCount = m_localBookmarks.size() - 1;
+                        m_lastKnownLocation = viewBookmarkVisitor.m_lastKnownLocation;
+                    }
+
+                    // once loaded we can remove the data from the settings registry.
+                    registry->Remove(bookmarkKey + "/");
+                    return visitedViewBookmarks;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    AZStd::optional<ViewBookmark> ViewBookmarkLoader::GetBookmarkAtIndex(int index, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case StorageMode::Shared:
+            AZ_Assert(false, "Shared Bookmark Component functionality not implemented.");
+            return AZStd::optional<ViewBookmark>();
+        case StorageMode::Local:
+
+            if (index >= 0 && index < m_localBookmarks.size())
+            {
+                return AZStd::optional<ViewBookmark>(m_localBookmarks.at(index));
+            }
+            AZ_Warning("ViewBookmarkLoader", false, "Couldn't load View Bookmark from file.");
+            return AZStd::optional<ViewBookmark>();
+        case StorageMode::Invalid:
+        default:
+            return AZStd::optional<ViewBookmark>();
+        }
+    }
+
+    bool ViewBookmarkLoader::RemoveBookmarkAtIndex(int index, const StorageMode mode)
+    {
+        switch (mode)
+        {
+        case StorageMode::Shared:
+            AZ_Assert(false, "Shared Bookmark Component functionality not implemented.");
+            return false;
+        case StorageMode::Local:
+            return RemoveLocalBookmarkAtIndex(index);
+        case StorageMode::Invalid:
+        default:
+            AZ_Warning("ViewBookmarkLoader", false, "Couldn't find ViewBookmarkComponent");
+            return false;
+        }
+    }
+
+    AZStd::optional<ViewBookmark> ViewBookmarkLoader::LoadLastKnownLocation() const
+    {
+        return m_lastKnownLocation;
+    }
+
+    LocalViewBookmarkComponent* ViewBookmarkLoader::RetrieveLocalViewBookmarkComponent()
+    {
+        AZ::EntityId levelEntityId;
+        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+            levelEntityId, &AzToolsFramework::ToolsApplicationRequests::GetCurrentLevelEntityId);
```

Comment:
question: @AMZN-daimini this is the call there might be a better alternative to use to get the 'root' prefab (which afaik doubles as the 'level' entity, but I could be wrong). Thanks!

Reply:
We could use `GetRootPrefabInstance()` in `\AzToolsFramework\Entity\PrefabEditorEntityOwnershipInterface.h`.
But honestly I'd be up for leaving it like this for now and actually bringing up a broader discussion about prefab editing / moving levels to a Gem / prefab contracts with the prefab team.